From 0f7eb01d7fe5fbd5b4dd40ebb2891829deaac983 Mon Sep 17 00:00:00 2001
From: Ping-Ke Shih <pkshih@realtek.com>
Date: Thu, 21 Apr 2022 20:08:54 +0800
Subject: [PATCH 470/515] rtw89: ser: re-enable interrupt in threadfn if
 under_recovery

Normally, we re-enable interrupt by napi_poll, but for this special
situation, we must turn it on immediately because napi_poll isn't
scheduled.

Signed-off-by: Ping-Ke Shih <pkshih@realtek.com>
Signed-off-by: Kalle Valo <kvalo@kernel.org>
Link: https://lore.kernel.org/r/20220421120903.73715-6-pkshih@realtek.com
---
 drivers/net/wireless/realtek/rtw89/pci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/realtek/rtw89/pci.c b/drivers/net/wireless/realtek/rtw89/pci.c
index e8bcecbe77e1..ad3db5aa890c 100644
--- a/drivers/net/wireless/realtek/rtw89/pci.c
+++ b/drivers/net/wireless/realtek/rtw89/pci.c
@@ -743,7 +743,7 @@ static irqreturn_t rtw89_pci_interrupt_threadfn(int irq, void *dev)
 		rtw89_ser_notify(rtwdev, rtw89_mac_get_err_status(rtwdev));
 
 	if (unlikely(rtwpci->under_recovery))
-		return IRQ_HANDLED;
+		goto enable_intr;
 
 	if (unlikely(rtwpci->low_power)) {
 		rtw89_pci_low_power_interrupt_handler(rtwdev);
-- 
2.35.3

